odoo/api.py

```diff
@@ -1017,167 +991,86 @@ def update(self, records, field, values, dirty=False, check_dirty=True):
         :param check_dirty: whether updating a dirty field without making it
             dirty must raise an exception
         """
-        if field.translate:
-            lang = records.env.lang or 'en_US'
-            field_cache = self._get_field_cache(records, field)
-            cache_values = []
-            for id_, value in zip(records._ids, values):
-                if value is None:
-                    cache_values.append(None)
-                else:
-                    cache_value = field_cache.get(id_) or {}
-                    cache_value[lang] = value
-                    cache_values.append(cache_value)
-            values = cache_values
-
-        self.update_raw(records, field, values, dirty, check_dirty)
+        field_cache = self._set_field_cache(field, context)
+        if updater:
+            for id_, val in zip(ids, values):
+                updater(field_cache, id_, val)
+        else:
+            field_cache.update(zip(ids, values))
```

Comment:
If `updater` takes the `field_cache` dict and the pairs `(id, val)`, it could be simplified as
```suggestion
        if updater is None:
            updater = dict.update
        updater(field_cache, zip(ids, values))
```
(Not sure.)